nemo/collections/asr/parts/preprocessing/perturb.py

```diff
@@ -583,6 +570,204 @@ def perturb_with_foreground_noise(self, data, noise, data_rms=None, max_noise_du
+        self._norm_to_db = norm_to_db
+        self._snr_samples = snr_samples if isinstance(snr_samples, list) and len(snr_samples) > 0 else None
+        self._epsilon = epsilon
+
+    @property
+    def orig_sr(self):
+        return self._orig_sr
+
+    def read_one_audiosegment(self, target_sr):
+        if self._tarred_audio:
+            if self._data_iterator is None:
+                raise TypeError("Expected valid iterator but got None")
+            try:
+                audio_file, file_id, manifest_entry = next(self._data_iterator)
+            except StopIteration:
+                self._data_iterator = iter(self._audiodataset)
+                audio_file, file_id, manifest_entry = next(self._data_iterator)
+
+            offset = 0 if manifest_entry.offset is None else manifest_entry.offset
+            duration = 0 if manifest_entry.duration is None else manifest_entry.duration
+
+        else:
+            audio_record = random.sample(self._manifest.data, 1)[0]
+            audio_file = audio_record.audio_file
+            offset = 0 if audio_record.offset is None else audio_record.offset
+            duration = 0 if audio_record.duration is None else audio_record.duration
+
+        return AudioSegment.from_file(audio_file, target_sr=target_sr, offset=offset, duration=duration)
+
+    def perturb(self, data, ref_mic=0):
+        """
+        Args:
+            data (AudioSegment): audio data
+            ref_mic (int): reference mic index for scaling multi-channel audios
+        """
+
+        noise = self.read_one_audiosegment(data.sample_rate)
+        while noise.duration < 1:
+            noise = self.read_one_audiosegment(data.sample_rate)
+
+        self.perturb_with_input_noise(data, noise, ref_mic=ref_mic, norm_to_db=self._norm_to_db)
+
+    def snr_mixer(self, clean, noise, snr, norm_to_db=-25.0):
+        rmsclean = (clean ** 2).mean(axis=0) ** 0.5
+        rmsclean = np.where(np.isclose(rmsclean, 0), self._epsilon, rmsclean)
+        scalarclean = 10 ** (norm_to_db / 20) / rmsclean
+        clean = clean * scalarclean
+        rmsclean = (clean ** 2).mean(axis=0) ** 0.5
+
+        rmsnoise = (noise ** 2).mean(axis=0) ** 0.5
+        rmsnoise = np.where(np.isclose(rmsnoise, 0), self._epsilon, rmsnoise)
+        scalarnoise = 10 ** (norm_to_db / 20) / rmsnoise
+        noise = noise * scalarnoise
+        rmsnoise = (noise ** 2).mean(axis=0) ** 0.5
+        rmsnoise = np.where(np.isclose(rmsnoise, 0), self._epsilon, rmsnoise)
+
+        # Set the noise level for a given SNR
+        noisescalar = np.sqrt(rmsclean / (10 ** (snr / 20)) / rmsnoise)
+        noisenewlevel = noise * noisescalar
+        noisyspeech = clean + noisenewlevel
+
+        return clean, noisenewlevel, noisyspeech
+
+    def norm_audio_to_db(self, x, norm_to_db):
+        rms = (x ** 2).mean(axis=0) ** 0.5
+        rms = np.where(np.isclose(rms, 0), self._epsilon, rms)
+        scalar = 10 ** (norm_to_db / 20.0) / rms
+        return x * scalar
```

Comment:
Yes I know, but in my opinion, there is no need to do an import for a simple one-liner of code. For example, db2mag is just "return 10 ** (db / 20)". That kind of simple calculation is not worth an import statement, it becomes neater and easier to just do this calculation as I have already written it. My approach also makes it easier for future people who are working on this class to just easily see all the math in one place, rather than obfuscate it behind a different file all for the sake of an exponential and division operation.